samples/settings/boards/esp32.conf

```diff
@@ -0,0 +1,30 @@
+CONFIG_WIFI=y
+CONFIG_WIFI_ESP32=y
+CONFIG_HEAP_MEM_POOL_SIZE=131072
+
+CONFIG_NET_L2_ETHERNET=y
+
+CONFIG_NET_DHCPV4=y
+
+CONFIG_NET_CONFIG_LOG_LEVEL_DBG=y
+CONFIG_NET_CONFIG_SETTINGS=y
+CONFIG_NET_CONFIG_NEED_IPV4=y
+
+CONFIG_MBEDTLS_ENTROPY_ENABLED=y
+CONFIG_MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED=y
+CONFIG_MBEDTLS_ECP_ALL_ENABLED=y
+
+
+CONFIG_ESP32_WIFI_STA_AUTO=y
+CONFIG_GOLIOTH_SAMPLE_WIFI=n
+
+CONFIG_MCUMGR=n
+CONFIG_MCUMGR_SMP_SHELL=n
+CONFIG_MCUMGR_CMD_CFG_MGMT=n
+CONFIG_MCUMGR_CMD_OS_MGMT=n
```

Comment:
Why do you disable MCUMGR?

Reply:
@alvarowolfx, removing all the 4 lines about mcumgr, i got an error when building.

But at the same time, you think it's not needed.

If i let only the first line 'CONFIG_MCUMGR=n', the build finishes ok.

what's the best way to proceed with this fix?

The error is
```
❯ west build --pristine -b esp32 samples/settings
-- west build: making build dir /home/miguel/Golioth/Codes/zephyr-setting/zephyr-sdk/build pristine
-- west build: generating a build system
Including boilerplate (Zephyr base): /home/miguel/Golioth/Codes/zephyrproject/zephyr/cmake/app/boilerplate.cmake
-- Application: /home/miguel/Golioth/Codes/zephyr-setting/zephyr-sdk/samples/settings
-- Zephyr version: 2.7.99 (/home/miguel/Golioth/Codes/zephyrproject/zephyr), build: v2.7.99-3661-gd51a67b7c098
-- Found Python3: /home/miguel/Golioth/Codes/zephyrproject/.venv/bin/python3 (found suitable exact version "3.8.10") found components: Interpreter 
-- Found west (found suitable version "0.12.0", minimum required is "0.7.1")
-- Board: esp32
-- Cache files will be written to: /home/miguel/.cache/zephyr
-- Found dtc: /usr/bin/dtc (found suitable version "1.5.0", minimum required is "1.4.6")
-- Found BOARD.dts: /home/miguel/Golioth/Codes/zephyrproject/zephyr/boards/xtensa/esp32/esp32.dts
-- Generated zephyr.dts: /home/miguel/Golioth/Codes/zephyr-setting/zephyr-sdk/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /home/miguel/Golioth/Codes/zephyr-setting/zephyr-sdk/build/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: /home/miguel/Golioth/Codes/zephyr-setting/zephyr-sdk/build/zephyr/include/generated/device_extern.h
-- Including generated dts.cmake file: /home/miguel/Golioth/Codes/zephyr-setting/zephyr-sdk/build/zephyr/dts.cmake

warning: LOG_STRDUP_BUF_COUNT (defined at subsys/logging/Kconfig.processing:133) was assigned the
value '40' but got the value ''. Check these unsatisfied dependencies: LOG1_DEFERRED (=n). See
http://docs.zephyrproject.org/latest/reference/kconfig/CONFIG_LOG_STRDUP_BUF_COUNT.html and/or look
up LOG_STRDUP_BUF_COUNT in the menuconfig/guiconfig interface. The Application Development Primer,
Setting Configuration Values, and Kconfig - Tips and Best Practices sections of the manual might be
helpful too.

Parsing /home/miguel/Golioth/Codes/zephyr-setting/zephyr-sdk/samples/settings/Kconfig
Loaded configuration '/home/miguel/Golioth/Codes/zephyrproject/zephyr/boards/xtensa/esp32/esp32_defconfig'
Merged configuration '/home/miguel/Golioth/Codes/zephyr-setting/zephyr-sdk/samples/settings/prj.conf'
Merged configuration '/home/miguel/Golioth/Codes/zephyr-setting/zephyr-sdk/samples/settings/boards/esp32.conf'
Configuration saved to '/home/miguel/Golioth/Codes/zephyr-setting/zephyr-sdk/build/zephyr/.config'
Kconfig header saved to '/home/miguel/Golioth/Codes/zephyr-setting/zephyr-sdk/build/zephyr/include/generated/autoconf.h'
-- Found toolchain: espressif (/home/miguel/.espressif/tools/zephyr)
-- The C compiler identification is GNU 8.4.0
-- The CXX compiler identification is GNU 8.4.0
-- The ASM compiler identification is GNU
-- Found assembler: /home/miguel/.espressif/tools/zephyr/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc
CMake Warning at /home/miguel/Golioth/Codes/zephyrproject/zephyr/CMakeLists.txt:763 (message):
  No SOURCES given to Zephyr library: subsys__net__pkt_filter

  Excluding target from build.


CMake Warning at /home/miguel/Golioth/Codes/zephyrproject/zephyr/CMakeLists.txt:763 (message):
  No SOURCES given to Zephyr library: drivers__ethernet

  Excluding target from build.


-- Configuring done
-- Generating done
-- Build files have been written to: /home/miguel/Golioth/Codes/zephyr-setting/zephyr-sdk/build
-- west build: building application
[1/441] Preparing syscall dependency handling

[422/441] Performing configure step for 'EspIdfBootloader'
CMake Warning at /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/tools/cmake/idf.cmake:24 (message):
  IDF_PATH environment variable not found.  Setting IDF_PATH to
  '/home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif'.
Call Stack (most recent call first):
  /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/tools/cmake/project.cmake:7 (include)
  CMakeLists.txt:34 (include)


-- Found Git: /usr/bin/git (found version "2.25.1") 
-- The C compiler identification is GNU 8.4.0
-- The CXX compiler identification is GNU 8.4.0
-- The ASM compiler identification is GNU
-- Found assembler: /home/miguel/.espressif/tools/zephyr/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /home/miguel/.espressif/tools/zephyr/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /home/miguel/.espressif/tools/zephyr/xtensa-esp32-elf/bin/xtensa-esp32-elf-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Building ESP-IDF components for target esp32
-- Project sdkconfig file /home/miguel/Golioth/Codes/zephyr-setting/zephyr-sdk/build/esp-idf/build/sdkconfig
-- Adding linker script /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/esp32/ld/esp32.peripherals.ld
-- Adding linker script /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/esp_rom/esp32/ld/esp32.rom.ld
-- Adding linker script /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/esp_rom/esp32/ld/esp32.rom.api.ld
-- Adding linker script /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/esp_rom/esp32/ld/esp32.rom.libgcc.ld
-- Adding linker script /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/esp_rom/esp32/ld/esp32.rom.newlib-funcs.ld
-- Adding linker script /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/bootloader/subproject/main/ld/esp32/bootloader.ld
-- Adding linker script /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/bootloader/subproject/main/ld/esp32/bootloader.rom.ld
-- Components: bootloader bootloader_support efuse esp32 esp_common esp_hw_support esp_rom esp_system esptool_py hal log main micro-ecc newlib partition_table soc spi_flash xtensa
-- Component paths: /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/bootloader /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/bootloader_support /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/efuse /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/esp32 /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/esp_common /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/esp_hw_support /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/esp_rom /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/esp_system /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/esptool_py /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/hal /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/log /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/bootloader/subproject/main /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/bootloader/subproject/components/micro-ecc /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/newlib /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/partition_table /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/soc /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/spi_flash /home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/xtensa
-- Configuring done
-- Generating done
-- Build files have been written to: /home/miguel/Golioth/Codes/zephyr-setting/zephyr-sdk/build/esp-idf/build/bootloader
[423/441] Performing build step for 'EspIdfBootloader'
[1/85] Generating project_elf_src_esp32.c
[2/85] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/soc_include_legacy_warn.c.obj
[3/85] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/i2c_periph.c.obj
[4/85] Building C object esp-idf/hal/CMakeFiles/__idf_hal.dir/mpu_hal.c.obj
[5/85] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/lldesc.c.obj
[6/85] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/dac_periph.c.obj
[7/85] Building C object CMakeFiles/bootloader.elf.dir/project_elf_src_esp32.c.obj
[8/85] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/adc_periph.c.obj
[9/85] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/memory_layout_utils.c.obj
[10/85] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/gpio_periph.c.obj
[11/85] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/interrupts.c.obj
[12/85] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/rmt_periph.c.obj
[13/85] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/ledc_periph.c.obj
[14/85] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/i2s_periph.c.obj
[15/85] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/rtc_io_periph.c.obj
[16/85] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/soc_memory_layout.c.obj
[17/85] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/pcnt_periph.c.obj
[18/85] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/sdmmc_periph.c.obj
[19/85] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/sdio_slave_periph.c.obj
[20/85] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/sigmadelta_periph.c.obj
[21/85] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/timer_periph.c.obj
[22/85] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/uart_periph.c.obj
[23/85] Building C object esp-idf/hal/CMakeFiles/__idf_hal.dir/wdt_hal_iram.c.obj
[24/85] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/touch_sensor_periph.c.obj
[25/85] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/spi_periph.c.obj
[26/85] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_mem.c.obj
[27/85] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_random.c.obj
[28/85] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_clock_init.c.obj
[29/85] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_common_loader.c.obj
[30/85] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_random_esp32.c.obj
[31/85] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/flash_encrypt.c.obj
[32/85] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_common.c.obj
[33/85] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/secure_boot.c.obj
[34/85] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/flash_partitions.c.obj
[35/85] Building C object esp-idf/spi_flash/CMakeFiles/__idf_spi_flash.dir/esp32/spi_flash_rom_patch.c.obj
[36/85] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_flash.c.obj
[37/85] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/flash_qio_mode.c.obj
[38/85] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_clock_loader.c.obj
[39/85] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_console_loader.c.obj
[40/85] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_init.c.obj
[41/85] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_efuse_esp32.c.obj
[42/85] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_flash_config_esp32.c.obj
[43/85] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_console.c.obj
[44/85] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_panic.c.obj
[45/85] Building C object esp-idf/efuse/CMakeFiles/__idf_efuse.dir/esp32/esp_efuse_table.c.obj
[46/85] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/esp32/bootloader_sha.c.obj
[47/85] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_utility.c.obj
[48/85] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/esp_image_format.c.obj
[49/85] Building C object esp-idf/efuse/CMakeFiles/__idf_efuse.dir/src/esp32/esp_efuse_api.c.obj
[50/85] Building C object esp-idf/efuse/CMakeFiles/__idf_efuse.dir/src/esp32/esp_efuse_fields.c.obj
[51/85] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/esp32/flash_encrypt.c.obj
[52/85] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/compare_set.c.obj
[53/85] Building C object esp-idf/efuse/CMakeFiles/__idf_efuse.dir/src/esp_efuse_fields.c.obj
[54/85] Building C object esp-idf/efuse/CMakeFiles/__idf_efuse.dir/src/esp32/esp_efuse_utility.c.obj
[55/85] Building C object esp-idf/efuse/CMakeFiles/__idf_efuse.dir/src/esp_efuse_api.c.obj
[56/85] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/esp32/bootloader_esp32.c.obj
[57/85] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/cpu_util.c.obj
[58/85] Building C object esp-idf/esp_rom/CMakeFiles/__idf_esp_rom.dir/patches/esp_rom_crc.c.obj
[59/85] Building C object esp-idf/efuse/CMakeFiles/__idf_efuse.dir/src/esp_efuse_utility.c.obj
[60/85] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/port/esp32/rtc_pm.c.obj
[61/85] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/port/esp32/rtc_clk_init.c.obj
[62/85] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/port/esp32/rtc_wdt.c.obj
[63/85] Building C object esp-idf/esp_rom/CMakeFiles/__idf_esp_rom.dir/patches/esp_rom_sys.c.obj
[64/85] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/port/esp32/rtc_init.c.obj
[65/85] Building C object esp-idf/esp_rom/CMakeFiles/__idf_esp_rom.dir/patches/esp_rom_uart.c.obj
[66/85] Building C object esp-idf/log/CMakeFiles/__idf_log.dir/log.c.obj
[67/85] Building C object esp-idf/micro-ecc/CMakeFiles/__idf_micro-ecc.dir/uECC_verify_antifault.c.obj
[68/85] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/port/esp32/rtc_time.c.obj
[69/85] Building C object esp-idf/log/CMakeFiles/__idf_log.dir/log_noos.c.obj
[70/85] Building C object esp-idf/main/CMakeFiles/__idf_main.dir/bootloader_start.c.obj
[71/85] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/port/esp32/rtc_sleep.c.obj
[72/85] Building C object esp-idf/log/CMakeFiles/__idf_log.dir/log_buffers.c.obj
[73/85] Linking C static library esp-idf/log/liblog.a
[74/85] Linking C static library esp-idf/esp_rom/libesp_rom.a
[75/85] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/port/esp32/rtc_clk.c.obj
[76/85] Linking C static library esp-idf/esp_hw_support/libesp_hw_support.a
[77/85] Linking C static library esp-idf/efuse/libefuse.a
[78/85] Linking C static library esp-idf/bootloader_support/libbootloader_support.a
[79/85] Linking C static library esp-idf/spi_flash/libspi_flash.a
[80/85] Linking C static library esp-idf/micro-ecc/libmicro-ecc.a
[81/85] Linking C static library esp-idf/soc/libsoc.a
[82/85] Linking C static library esp-idf/hal/libhal.a
[83/85] Linking C static library esp-idf/main/libmain.a
[84/85] Linking C executable bootloader.elf
[85/85] Generating binary image from built executable
esptool.py v3.1-dev
Merged 1 ELF section
Generated /home/miguel/Golioth/Codes/zephyr-setting/zephyr-sdk/build/esp-idf/build/bootloader/bootloader.bin
[431/441] Linking C executable zephyr/zephyr_pre0.elf
FAILED: zephyr/zephyr_pre0.elf zephyr/zephyr_pre0.map /home/miguel/Golioth/Codes/zephyr-setting/zephyr-sdk/build/zephyr/zephyr_pre0.map 
: && ccache /home/miguel/.espressif/tools/zephyr/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc   zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj -o zephyr/zephyr_pre0.elf  zephyr/CMakeFiles/offsets.dir/./arch/xtensa/core/offsets/offsets.c.obj  -fuse-ld=bfd  -Wl,-T  zephyr/linker_zephyr_pre0.cmd  -Wl,-Map=/home/miguel/Golioth/Codes/zephyr-setting/zephyr-sdk/build/zephyr/zephyr_pre0.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/xtensa/core/libarch__xtensa__core.a  zephyr/lib/libc/minimal/liblib__libc__minimal.a  zephyr/lib/posix/liblib__posix.a  zephyr/subsys/mgmt/mcumgr/libsubsys__mgmt__mcumgr.a  zephyr/subsys/net/libsubsys__net.a  zephyr/subsys/net/l2/ethernet/libsubsys__net__l2__ethernet.a  zephyr/subsys/net/ip/libsubsys__net__ip.a  zephyr/subsys/net/lib/config/libsubsys__net__lib__config.a  zephyr/subsys/net/lib/dns/libsubsys__net__lib__dns.a  zephyr/subsys/random/libsubsys__random.a  zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a  zephyr/drivers/clock_control/libdrivers__clock_control.a  zephyr/drivers/console/libdrivers__console.a  zephyr/drivers/gpio/libdrivers__gpio.a  zephyr/drivers/i2c/libdrivers__i2c.a  zephyr/drivers/wifi/libdrivers__wifi.a  zephyr/drivers/hwinfo/libdrivers__hwinfo.a  zephyr/drivers/flash/libdrivers__flash.a  zephyr/drivers/serial/libdrivers__serial.a  zephyr/drivers/entropy/libdrivers__entropy.a  zephyr/drivers/timer/libdrivers__timer.a  modules/golioth/mgmt/mcumgr/libMCUMGR_CFG.a  modules/qcbor/lib..__modules__lib__qcbor__zephyr.a  modules/mbedtls/libmodules__mbedtls.a  modules/tinycbor/lib..__modules__lib__tinycbor__zephyr.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  -L"/home/miguel/.espressif/tools/zephyr/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/8.4.0"  -L/home/miguel/Golioth/Codes/zephyr-setting/zephyr-sdk/build/zephyr  -lgcc  zephyr/arch/common/libisr_tables.a  -no-pie  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn  -T/home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/zephyr/esp32/src/linker/esp32.rom.alias.ld  -T/home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/zephyr/esp32/../../components/esp_rom/esp32/ld/esp32.rom.ld  -T/home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/zephyr/esp32/../../components/esp_rom/esp32/ld/esp32.rom.api.ld  -T/home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/zephyr/esp32/../../components/esp_rom/esp32/ld/esp32.rom.libgcc.ld  -T/home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/zephyr/esp32/../../components/esp_rom/esp32/ld/esp32.rom.redefined.ld  -T/home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/zephyr/esp32/../../components/esp32/ld/esp32.peripherals.ld  -Wl,--wrap=k_calloc  -Wl,--wrap=k_malloc  -lnet80211  -lcore  -lpp  -lcoexist  -lphy  -lrtc  -lgcc  -L/home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/zephyr/esp32/../../components/esp32  -L/home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/zephyr/esp32/../../components/esp_wifi/lib/esp32  -Wl,--defsym,phy_printf=printf  -Wl,--defsym,net80211_printf=printf  -Wl,--defsym,coexist_printf=printf && cd /home/miguel/Golioth/Codes/zephyr-setting/zephyr-sdk/build/zephyr && /usr/local/bin/cmake -E echo
/home/miguel/.espressif/tools/zephyr/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.bfd: zephyr/libzephyr.a(base64.c.obj): in function `base64_encode':
/home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/wpa_supplicant/src/utils/base64.c:166: multiple definition of `base64_encode'; zephyr/libzephyr.a(base64.c.obj):/home/miguel/Golioth/Codes/zephyrproject/zephyr/lib/os/base64.c:62: first defined here
/home/miguel/.espressif/tools/zephyr/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.bfd: zephyr/libzephyr.a(base64.c.obj): in function `base64_decode':
/home/miguel/Golioth/Codes/zephyrproject/modules/hal/espressif/components/wpa_supplicant/src/utils/base64.c:188: multiple definition of `base64_decode'; zephyr/libzephyr.a(base64.c.obj):/home/miguel/Golioth/Codes/zephyrproject/zephyr/lib/os/base64.c:126: first defined here
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/local/bin/cmake --build /home/miguel/Golioth/Codes/zephyr-setting/zephyr-sdk/build
```

Reply:
I believe @szczys ran into similar problems in the past with multiple definitions of `base64_decode`. He's on vacation today, but he might be able to help tomorrow.

Reply:
Just tried newest Zephyr (tip of main) and the problem seems to be not fixed yet. Both wpa_supplicant (which ESP uses) and mcumgr use base64 functions and there is a function name conflict. Solving this doesn't seem to be trivial.

Question is what do we do about it. At this point this sample explicitly assumes that mcumgr will be used. This is the head of README.rst file:
```
Golioth Settings sample
#######################

Overview
********

This sample application demonstrates how to enable Zephyr settings subsystem and
use it for storing Golioth credentials. Additionally it shows how to provision
these credentials using ``mcumgr`` CLI.
```
which says that we expect users to run `mcumgr`. We cannot just disable mcumgr for esp32 target, as it no longer makes sense to even include this platform configuration in this sample. We can decide to broaden the scope of this sample (to support more than mcumgr), but that requires to add more information to README.rst.

Reply:
You could add a post-build step to use objcopy to prefix symbols in `wpa_supplicant.o`, similar to what's described here: https://stackoverflow.com/a/6940389

But it's a huge hack. On the other hand, I don't know how else we can resolve it without having an upstream fix.